src/StringTools/InternableString.cs

```diff
@@ -304,28 +304,59 @@ public override string ToString()
         /// <returns>A stable hashcode of the string represented by this instance.</returns>
         public override unsafe int GetHashCode()
         {
-            int hashCode = 5381;
+            uint hash = (5381 << 16) + 5381;
```

Comment:
slightly off-topic: Wouldn't caching this string's hashcode result in a net improvement?

Reply:
`GetHashCode` runs only once for a given string unless the caller calls `SpanBasedStringBuilder.ToString()` multiple times on the same instance without mutating it between the calls. Same as calling `StringBuilder.ToString()` multiple times, it is technically possible but enough of an anti-pattern that the implementation does not cache the result.

More on running `GetHashCode` only once: When a string is added to the weak cache, its hash code is used as a key in a dictionary. So it's not calculated on each look-up because the look-up is done based on the hash code and not the string itself. When we're looking for a string in the weak cache, we calculate its hash code once per lookup and, as argued above, there should not be more than one look-up for the same string.